src/traversals/bfs.jl

```diff
@@ -65,23 +65,26 @@ For vertices in disconnected components the default distance is -1.
 function gdistances!(g::AbstractGraph, source, dists)
     T = eltype(g)
     n = nv(g)
-    fill!(dists, -1)
+    fill!(dists, typemax(T))
+    seen = zeros(Bool, n)
```

Comment:
This is an extra O(nv) storage from using the dists array with -1.

Reply:
Oh, because of the `seen` - don't worry, that will be a bitvector very soon. It drops the time on my large graph from 70s to 45s.